docs/running-tests/from-local-system.md

```diff
@@ -4,69 +4,86 @@ The tests are designed to be run from your local computer.
 
 ## System Setup
 
-The test environment requires [Python 2.7+](http://www.python.org/downloads)
-(but not Python 3.x).
+Running the tests requires `python`, `pip` and `virtualenv`, as well as updating
+the system `hosts` file.
 
-On Windows, be sure to add the Python directory (`c:\python2x`, by default) to
-your `%Path%` [Environment Variable](http://www.computerhope.com/issues/ch000549.htm),
-and read the [Windows Notes](#windows-notes) section below.
+Note that Python 2.7 is required. Using Python 3 is not supported.
 
-<!--
-  There does not appear to be a cross-platform means of installing `pip`.
-  https://github.com/web-platform-tests/wpt/pull/16670
--->
+The required setup is different depending on your operating system.
 
-Install `pip`. On many systems, this can be achieved with the command `python
--m ensurepip`. If this is not possible, use your system's package manager to
-install the `python-pip` package.
+### Linux Setup
 
-Next, install `virtualenv` using the following command:
+If not already present, use the system package manager to install `python`,
+`pip` and `virtualenv`.
+
+On Debian or Ubuntu:
 
 ```bash
-pip install virtualenv
+sudo apt-get install python python-pip virtualenv
 ```
 
-To get the tests running, you need to set up the test domains in your
-[`hosts` file](http://en.wikipedia.org/wiki/Hosts_%28file%29%23Location_in_the_file_system).
-
-The necessary content can be generated with `./wpt make-hosts-file`; on
-Windows, you will need to preceed the prior command with `python` or
-the path to the Python binary (`python wpt make-hosts-file`).
-
-For example, on most UNIX-like systems, you can setup the hosts file with:
+On Fedora or Red Hat:
 
 ```bash
-./wpt make-hosts-file | sudo tee -a /etc/hosts
+sudo dnf install python2 python2-pip python2-virtualenv
 ```
 
-And on Windows (this must be run in a PowerShell session with Administrator privileges):
+### macOS Setup
+
+The system-provided Python can be used, while `pip` and `virtualenv` can be
+installed for the user only:
 
 ```bash
-python wpt make-hosts-file | Out-File %SystemRoot%\System32\drivers\etc\hosts -Encoding ascii -Append
+python -m ensurepip --user
+export PATH="$PATH:$HOME/Library/Python/2.7/bin"
+pip install --user virtualenv
 ```
 
-If you are behind a proxy, you also need to make sure the domains above are
-excluded from your proxy lookups.
+To make the `PATH` change persistent, add it to your `~/.profile` file.
 
-[The Ahem font](../writing-tests/ahem) is used to test precise rendering
-behavior. This font should be loaded as a web font in tests, using the
-`/fonts/ahem.css` stylesheet, as follows:
+See also [additional setup required to run Safari](safari).
 
-```
-<link rel="stylesheet" type="text/css" href="/fonts/ahem.css" />
-```
+### Windows Setup
```

Comment:
@stephenmcgruer can you review the overall PR and try the steps for Windows?

Reply:
https://dev.azure.com/web-platform-tests/wpt/_build/results?buildId=35625 is evidence that I've gotten the path right and that both `pip` and `virtualenv` are in C:\Python27\Scripts.

Reply:
Actually in the interest of getting something landed today I've asked @Hexcles to do the overall review.

I'm fairly confident that these steps are more complete and correct than the previous ones, because this matches what we do in CI.